docs/modeling/models.rst

```diff
@@ -239,7 +239,15 @@ with `~astropy.modeling.custom_model` or as a `~astropy.modeling.core.CompoundMo
     ...
     >>> model = Ellipsoid3D()
     >>> model.bounding_box
-    ((-4.0, 4.0), (-3.0, 3.0), (-2.0, 2.0))
+    BoundingBox(
```

Comment:
I don't know how I feel about this. This is going to break my downstream packages big time, won't it?

https://github.com/spacetelescope/synphot_refactor/blob/3cc67bc7142f3c3b399c8d5c78cda06a775156fa/synphot/models.py#L64

Reply:
This new bounding box will function just fine with descriptions so long as one respects the bounding_box interface in modeling. That is either using the bounding_box setter or defining a bounding_box method (as your link references, note that there are astropy models like [`Gaussian1D`](https://github.com/astropy/astropy/blob/0b81dfdf0fe2bf41ec15a4d505d84ce53143adf6/astropy/modeling/functional_models.py#L125-L155) and [`Gaussian2D`](https://github.com/astropy/astropy/blob/0b81dfdf0fe2bf41ec15a4d505d84ce53143adf6/astropy/modeling/functional_models.py#L328-L364) which function in the same way). All of the necessary changes have been made to `Model` so the previously established methods of defining bounding boxes still work by performing a conversion to the "new" bounding box if necessary.

As far as the change to the documentation referenced here, I simply added a more descriptive `__repr__` to the new bounding box so that users could more easily determine how their bounding box was being interpreted (its clear what is `x`, `y` and `z` input intervals).

Reply:
Okay... I guess I'll just have to see how many things break after this is merged, but at least I know how to bother... 😸

Reply:
I have a set of changes for gwcs in place to accept this bounding box once it gets merged (mostly to remove the need for constant flipping of the bounding box order) and one tiny change to jwst (a warning filter needs to be added to a single test).

Reply:
Can you please xref those PRs here for future reference? Thanks!